### PR TITLE
Master - ITSMConfigurationManagement commands

### PR DESCRIPTION
**Pull request by s7design**
Hi @UdoBretz 

There are console commands for  ITSMConfigurationManagement.
I have written the first time console commands, and your follow up will be welcome. 

If you have any suggestion let me know. 

Regards
Zoran

### CODE REVIEW

File: Kernel/System/Console/Command/Admin/ITSM/Delete.pm [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    );
+    $Self->AddOption(
+        Name        => 'deploymentstate',
+        Description => "Delete all config items with this deployment state (ONLY TOGETHER with the --class parameter)",
+        Required    => 0,
+        HasValue    => 1,
+        ValueRegex  => qr/.*/smx,
+    );
+    $Self->AddOption(
+        Name        => 'configitem_number',
```
Comment (mgruner) [inline Kernel/System/Console/Command/Admin/ITSM/Delete.pm]: I'd recommend configitem-number as the resulting option will be --configitem-number.

File: Kernel/System/Console/Command/Admin/ITSM/Delete.pm [hunk 2]
```diff
@@ -0,0 +1,258 @@
+# --
+# Copyright (C) 2001-2015 OTRS AG, http://otrs.com/
+# --
+# This software comes with ABSOLUTELY NO WARRANTY. For details, see
+# the enclosed file COPYING for license information (AGPL). If you
+# did not receive this file, see http://www.gnu.org/licenses/agpl.txt.
+# --
+
+package Kernel::System::Console::Command::Admin::ITSM::Delete;
```
Comment (mgruner) [inline Kernel/System/Console/Command/Admin/ITSM/Delete.pm]: If this is only about config items, the name should be Admin::ITSM::ConfigItem::Delete.

File: Kernel/System/Console/Command/Admin/ITSM/IncidentStateRecalculate.pm [hunk 3]
```diff
@@ -0,0 +1,117 @@
+# --
+# Copyright (C) 2001-2015 OTRS AG, http://otrs.com/
+# --
+# This software comes with ABSOLUTELY NO WARRANTY. For details, see
+# the enclosed file COPYING for license information (AGPL). If you
+# did not receive this file, see http://www.gnu.org/licenses/agpl.txt.
+# --
+
+package Kernel::System::Console::Command::Admin::ITSM::IncidentStateRecalculate;
```
Comment (mgruner) [inline Kernel/System/Console/Command/Admin/ITSM/IncidentStateRecalculate.pm]: Admin::ITSM::IncidentState::Recalculate

File: Kernel/System/Console/Command/Admin/ITSM/ListDuplicates.pm [hunk 4]
```diff
@@ -0,0 +1,231 @@
+# --
+# Copyright (C) 2001-2015 OTRS AG, http://otrs.com/
+# --
+# This software comes with ABSOLUTELY NO WARRANTY. For details, see
+# the enclosed file COPYING for license information (AGPL). If you
+# did not receive this file, see http://www.gnu.org/licenses/agpl.txt.
+# --
+
+package Kernel::System::Console::Command::Admin::ITSM::ListDuplicates;
```
Comment (mgruner) [inline Kernel/System/Console/Command/Admin/ITSM/ListDuplicates.pm]: Admin::ITSM::ConfigItem::ListDuplicates

File: Kernel/System/Console/Command/Admin/ITSM/ListDuplicates.pm [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    );
+    $Self->AddOption(
+        Name        => 'scope',
+        Description => "Define the scope for the uniqueness check (global|class)",
+        Required    => 0,
+        HasValue    => 1,
+        ValueRegex  => qr/(global|class)/smx,
+    );
+    $Self->AddOption(
+        Name        => 'allstates',
```
Comment (mgruner) [inline Kernel/System/Console/Command/Admin/ITSM/ListDuplicates.pm]: all-states maybe

**Comment by mgruner** [comment]
Hello Zoran, I reviewed the code a little. Looks nice so far. Do you like the new Console API? 

There is one more recommendation, if you look in the framework you'll see that I usually added a start message and a done message in most/all commands. Please do this for consistency too, if possible.
Also make sure to check out bin/otrs.Console.pl Dev::Code::Generate::ConsoleCommand. This also creates you stubs for unit  test files. Please add at least one basic test (just for exit code) for each new console command. If possible more (db checks etc.).

Regards, mg

**Comment by s7design** [comment]
Thanks @mgruner 

I like it a lot :) I was really happy while I had done on this. I have not done this before, and I left at the end after porting and writing Selenium test, it had looked like hard and strange before, but now I am getting to understand it, and it is so cool :)

Thanks for your comments which are very useful. I am changing all of that soon and than I am planning to make unit tests.

Regards
Zoran

**Comment by s7design** [comment]
And specially thanks for bin/otrs.Console.pl Dev::Code::Generate::ConsoleCommand it is awesome 
:+1:  :+1: :+1:  :+1: :+1:  :+1: 
:)

I have not known for this before. 

Regards
Zoran

**Comment by mgruner** [comment]
We can add more skeleton generators if needed.

**Comment by s7design** [comment]
Yes, I have already had some idea :)    

Maybe it could be useful to add more cases for options and arguments in Configure.

For example:

    #  $Self->AddOption(
    #     Name        => 'option',
    #     Description => "Describe this option.",
    #     Required    => 1,
    #     HasValue    => 1,
    #     ValueRegex  => qr/.*/smx,
    # );

    #  $Self->AddOption(
    #     Name        => 'multi_option',
    #     Description => "Describe this option.",
    #     Required    => 1,
    #     HasValue    => 1,
    #     ValueRegex  => qr/\d+/smx,
    #     Multiple    => 1,
    # );

And something like that.
Great, it is very helpful.

Regards
Zoran

**Comment by s7design** [comment]
Hi @UdoBretz 

I updated this PR. There are test for added command as well.
I tried to fiend the best test scenario for each of Command but if you have any suggestion please let me know.

Regards
Zoran

**Comment by UdoBretz** [comment]
Hello Zoran,

thank you very much! 

Best regards,
Udo

**Pull request closed without merging**